base_tier_validation/models/tier_validation.py

```diff
@@ -146,11 +153,20 @@ def _validate_tier(self, tiers=False):
         user_reviews = tier_reviews.filtered(
             lambda r: r.status in ('pending', 'rejected') and
             (self.env.user in r.reviewer_ids))
-        user_reviews.write({
-            'status': 'approved',
-            'done_by': self.env.user.id,
-            'reviewed_date': fields.Datetime.now(),
-        })
+        for user_review in user_reviews:
+            if user_review.approve_level:
+                tier_bf = tier_reviews.filtered(
+                    lambda r: r.sequence < user_review.sequence and
+                    r.status != 'approved'
+                )
+                if tier_bf:
```

Comment:
I am not sure it is good idea to have logic to check and raise error in a compute field.
Is it possible to do it other way?